### PR TITLE
Change .NET interop to internal, other type changes

### PR DESCRIPTION
**Pull request by thhous-msft**
## Description

Changes made by the .NET team in https://github.com/dotnet/runtime/pull/68288

### CODE REVIEW

**Review by nibanks — APPROVED** [review]

**Review by anrossi — DISMISSED** [review]

File: src/cs/lib/msquic_generated.cs [hunk 1]
```diff
@@ -2380,294 +2384,294 @@ public byte RESERVED
...
 
         [NativeTypeName("#define QUIC_PARAM_STREAM_PRIORITY 0x08000003")]
-        public const uint QUIC_PARAM_STREAM_PRIORITY = 0x08000003;
+        internal const uint QUIC_PARAM_STREAM_PRIORITY = 0x08000003;
 
         [NativeTypeName("#define QUIC_API_VERSION_2 2")]
-        public const uint QUIC_API_VERSION_2 = 2;
+        internal const uint QUIC_API_VERSION_2 = 2;
     }
 }
```
Comment (ManickaP) [inline src/cs/lib/msquic_generated.cs:2677]:
```suggestion
}
#pragma warning restore CS0649
```

File: src/cs/lib/msquic.cs [hunk 2]
```diff
@@ -10,12 +12,22 @@
 
 namespace Microsoft.Quic
 {
-    public unsafe partial struct QUIC_BUFFER
+#if PUBLIC_MSQUIC_INTEROP_ACCESS
```
Comment (ManickaP) [inline src/cs/lib/msquic.cs:15]: Is this still necessary when you have `InternalsVisibleTo` ? Or is there a reason for different approach in this file and `*_generate*.cs`?

**Review by ManickaP — APPROVED** [review]
Thanks!

File: src/cs/lib/msquic.cs (hunk 2 above)
Comment (nibanks) [reply src/cs/lib/msquic.cs:15]: How does `InternalsVisibleTo` work? That sounds like it might be a better solution.

Comment (ManickaP) [reply src/cs/lib/msquic.cs:15]: It makes types/fields/methods marked `internal` to look like `public` for assemblies named in the `InternalsVisibleTo` attribute. Disadvantage is that your interop lib must know the consumer lib/executable name beforehand.

Comment (thhous-msft) [reply src/cs/lib/msquic.cs:15]: Yeah I meant to revert these back. For our test specifically InternalsVisableTo is fine.

File: src/cs/lib/msquic_generated.cs (hunk 1 above)
Comment (thhous-msft) [reply src/cs/lib/msquic_generated.cs:2677]: Thats actually difficult to do in the generator. Is it needed to restore, even though its a generated file?

**Review by nibanks — APPROVED** [review]

**Review by raghavthind2005 — APPROVED** [review]

File: src/cs/lib/msquic_generated.cs (hunk 1 above)
Comment (ManickaP) [reply src/cs/lib/msquic_generated.cs:2677]: Not really, it should get restored at the end of the file anyway.

**Review by ManickaP — APPROVED** [review]
Thanks

**Pull request merged**